### ISSUE TITLE
[Bug]: missing select2 on sharing settings page

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#31570]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

![image](https://user-images.githubusercontent.com/277525/158181380-22216772-fa5b-4c07-abaf-6af1f620c5a4.png)

### Steps to reproduce

1. Setup Nextcloud locally (git master 371fa1c49a33ff1d03913b00446d109d03def1d7)
2. Go to the admin settings
3. Open the sharing section
4. Have a look at the section "Exclude groups from creating link shares:"

### Expected behavior

The selection UI element should be visible.

### Installation method

Other

### Operating system

Other

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.15.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_external: 1.16.0
  - files_sharing: 1.16.0
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - oauth2: 1.12.0
  - photos: 1.6.0
  - provisioning_api: 1.14.0
  - recommendations: 0.7.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.7.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
```

**Comment by PVince81** [comment]
seems not fully solved: https://github.com/nextcloud/server/issues/31589  @Pytal please have a look

**Comment by Pytal** [comment]
> seems not fully solved: #31589 @Pytal please have a look

@artonge

**Comment by Pytal** [comment]
Actually it's already closed
